atom/browser/common_web_contents_delegate.cc

```diff
@@ -205,8 +205,18 @@ void CommonWebContentsDelegate::SetOwnerWindow(
 
 void CommonWebContentsDelegate::ResetManagedWebContents(bool async) {
   if (async) {
-    base::ThreadTaskRunnerHandle::Get()->DeleteSoon(FROM_HERE,
-                                                    web_contents_.release());
+    // Browser context should be destroyed only after the WebContents,
+    // this is guaranteed in the sync mode by the order of declaration,
+    // in the async version we maintain a reference until the WebContents
+    // is destroyed.
+    base::ThreadTaskRunnerHandle::Get()->PostNonNestableTask(
+        FROM_HERE,
+        base::BindOnce(
+            [](scoped_refptr<AtomBrowserContext> browser_context,
+               void* web_contents) {
+              delete static_cast<content::WebContents*>(web_contents);
```

Comment:
Can you pass a `std::unique_ptr<content::WebContents>` instead of a `void*`? Converting from/to `void*` will lose the type check.

And it seems that `web_contents_` is a `brightray::InspectableWebContents` but it is being converted to `content::WebContents`?

Reply:
> And it seems that web_contents_ is a brightray::InspectableWebContents but it is being converted to content::WebContents?

Ah thats bad, thanks for catching it!